sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParseDriver.scala

```diff
@@ -28,7 +28,15 @@ import org.apache.spark.sql.AnalysisException
  * This is based on Hive's org.apache.hadoop.hive.ql.parse.ParseDriver
  */
 object ParseDriver extends Logging {
-  def parse(command: String, conf: ParserConf): ASTNode = {
+  def parsePlan(command: String, conf: ParserConf): ASTNode = parse(command, conf) { parser =>
```

Comment:
can you add some function doc for these 3 functions, and also cover what the differences are?